posts/hello-microservice-deployment-part-2.md

```diff
@@ -0,0 +1,358 @@
+
+Now say Webflix has a Pod that runs their recommendation service and they hit a spike in traffic. They would need to create a few new Pods to handle the traffic. A common approach is to use a Deployment for this.
+
+A Deployment specifies a Pod as well as rules around how that Pod is replicated. Eg it could say that we want 3 replicas of a pod containing recommend_service:1.2.0. It also does clever things like help with rolling out upgrades - if recommend_service:1.3.0 becomes available then you can instruct k8s to take down the recommend_service pods and bring up new pods in a sensible way.
+
+Deployments add a lot of power to Pods. To learn more about them I would like to refer you to the [official documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
+
+The last object I'd like to cover is the Service. Containers in a Pod can talk to each other as though they are on the same host. But you'll often need Pods to talk to other Pods, and you'll also need to expose some of your Pods to the big bad Internet because users outside your cluster will likely need to interact with your applications. This is achieved through Services. Services defines the network interfaces exposed by Pods. Again I would like to refer you to the [official documentation](https://kubernetes.io/docs/concepts/services-networking/service/) if you need more info.
+
+This was a very very brief introduction to K8s objects and their capabilities. Again there are [entire books](https://www.amazon.com/gp/product/1491935677/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=1491935677&linkCode=as2&tag=sheena0d-20&linkId=e5bf88f134212c4beba73c601598d7ff") written on the subject and if you want to become an expert on the topic I would urge you to do some reading on your own.
+
+## Practical: Deploying our application to Google cloud
+
+Now we know why a tool like K8s is useful... let's start using it. I've chosen to proceed by making use of Google's Kubernetes engine offering. You can use a different service provider if you want to, or even run your own cluster, or simulate a cluster using [minikube](https://kubernetes.io/docs/setup/minikube/). For the most part the tutorial will be exactly the same except for the initial cluster setup and the final cluster cleanup.
+
+### Set up our cluster
+
+Visit the Kubernetes engine page (https://console.cloud.google.com/projectselector/kubernetes). If you are new to Google cloud you'll need to sign in with your Google account. You can sign up for $300 worth of free credits while you are at it. If you aren't new to Google and have already used up your free credits then the cluster will cost money (not a lot though).
```

Comment:
Make this more concise. You could just say -- 
Visit the Kubernetes engine page (https://console.cloud.google.com/projectselector/kubernetes). If you are new to Google cloud you'll need to sign in with your Google account.